lib/metasploit/framework/ntds/account.rb

```diff
@@ -0,0 +1,178 @@
+        attr_accessor :nt_history_count
+        # @!attribute pass_date
+        #   @return [String] Human Readable Date for the last password change
+        attr_accessor :pass_date
+        # @!attribute pass_time
+        #   @return [String] Human Readable Time for the last password change
+        attr_accessor :pass_time
+        # @!attribute rid
+        #   @return [Fixnum] The Relative ID of the account
+        attr_accessor :rid
+        # @!attribute sid
+        #   @return [String] Byte String for the Account's SID
+        attr_accessor :sid
+
+        # @param raw_data [String] the raw 3948 byte string from the wire
+        # @raise [ArgumentErrror] if a 3948 byte string is not supplied
+        def initialize(raw_data)
+          raise ArgumentError, "No Data Supplied" unless raw_data.present?
+          raise ArgumentError, "Invalid Data" unless raw_data.length == 3948
+          data = raw_data.dup
+          @name = get_string(data,40)
+          @description = get_string(data,2048)
+          @rid = get_int(data)
+          @disabled = get_boolean(data)
+          @locked = get_boolean(data)
+          @no_pass = get_boolean(data)
+          @no_expire = get_boolean(data)
+          @expired = get_boolean(data)
+          @logon_count = get_int(data)
+          @nt_history_count = get_int(data)
+          @lm_history_count = get_int(data)
+          @expiry_date = get_string(data,30)
+          @logon_date =  get_string(data,30)
+          @logon_time = get_string(data,30)
+          @pass_date = get_string(data,30)
+          @pass_time = get_string(data,30)
+          @lm_hash = get_string(data,33)
+          @nt_hash = get_string(data,33)
+          @lm_history = get_hash_history(data)
+          @nt_history = get_hash_history(data)
+          @sid = data
+        end
+
+        # @return [String] String representation of the account data
+        def to_s
+          <<EOS
+#{@name} (#{@description})
+#{ntlm_hash}
+Password Expires: #{@expiry_date}
+Last Password Change: #{@pass_time} #{@pass_date}
+Last Logon: #{@logon_time} #{@logon_date}
+Logon Count: #{@logon_count}
+#{uac_string}
+Hash History:
+#{hash_history}
+EOS
+        end
+
+        # @return [String] the NTLM hash string for the current password
+        def ntlm_hash
+          "#{@name}:#{@rid}:#{@lm_hash}:#{@nt_hash}"
+        end
+
+        # @return [String] Each historical NTLM Hash on a new line
+        def hash_history
+          history_string = ''
+          @lm_history.each_with_index do | lm_hash, index|
+            history_string << "#{@name}:#{@rid}:#{lm_hash}:#{@nt_history[index]}\n"
+          end
+          history_string
+        end
+
+        private
+
+        def get_boolean(data)
+          get_int(data) == 1
+        end
+
+        def get_hash_history(data)
+          raw_history = data.slice!(0,792)
+          split_history = raw_history.scan(/.{1,33}/)
+          split_history.map!{ |hash| hash.gsub(/\x00/,'')}
```

Comment:
Are you swallowing NUL bytes because the encoding is actually utf16-le?  If that's the case you should be using the Ruby encode methods for Strings instead.